### ISSUE TITLE
Incorrect classification probabilities - Catboost

### ISSUE THREAD

**Issue by paul-cris** [curso-r/treesnip#29]
Hi, I'm running a classification using the Telco dataset but the outputted probabilities are loocking good

# A tibble: 6 x 2
  .pred_No .pred_Yes
     <dbl>     <dbl>
1    4.86     -3.86 
2    4.31     -3.31 
3    1.20     -0.198
4    3.85     -2.85 
5    4.10     -3.10 
6    0.358     0.642 

rm(list = ls(all.names=TRUE))

#------------------------------------------------------------------------------------------------------------#
### Preparing the data----

# Needed packages
library(tidymodels)
library(tune)
library(workflows)
library(tictoc)
library(treesnip)
library(foreach)
library(doParallel)
sessionInfo()

# Telco_customer <- read.csv("WA_Fn-UseC_-Telco-Customer-Churn.csv")
Telco_customer <- read.csv("https://raw.githubusercontent.com/treselle-systems/customer_churn_analysis/master/WA_Fn-UseC_-Telco-Customer-Churn.csv")
str(Telco_customer)
# Make copy of Telco_customer and drop the unneeded columns
data_set <- Telco_customer %>% dplyr::select(-"customerID")
# Rename the outcome variable (Churn in my case) to Target
data_in_scope <- data_set %>% plyr::rename(c("Churn" = "Target"))
# Drop rows with missing value(11 missing values, very small percentage of our total data)
data_in_scope <- data_set %>% plyr::rename(c("Churn" = "Target")) %>% drop_na()
# Check severity of class imbalance.
round(prop.table(table(data_in_scope$Target)), 2)

# Split the data into training and testing sets
set.seed(2020)
train_test_split <- data_in_scope %>%
  initial_split(prop = 0.8, strata = Target)

# Set recipie
recipie_num <- training(train_test_split) %>%
  recipe(Target ~.) %>%   # Fomula
  step_mutate(Target = as.factor(Target)) %>%
  step_dummy(all_nominal(), -Target) %>%          # convert nominal data into one or more numeric.
  step_corr(all_predictors()) %>%                 # remove variables that have large absolute
  # correlations with other variables.
  step_center(all_numeric(), -all_outcomes()) %>%  # normalize numeric data to have a mean of zero.
  step_scale(all_numeric(), -all_outcomes()) %>%   # normalize numeric data to have a standard deviation of one.
  prep()

# create CV object from training data
set.seed(2020)
data_folds <- recipes::bake(
  recipie_num,
  new_data = training(train_test_split)
) %>%
  rsample::vfold_cv(v = 10,
                    # repeats = 2,
                    strata = Target)

# create the final data
df.train <- as.data.frame(juice(recipie_num))
df.test <- as.data.frame(bake(recipie_num, new_data = testing(train_test_split)))

#------------------------------------------------------------------------------------------------------------#
### limit core usage
cl <- makeForkCluster(20)
registerDoParallel(cl)
foreach::getDoParWorkers()

# check which parameters are tunable
model <- parsnip::boost_tree(min_n = tune(),
                             learn_rate = tune(),
                             tree_depth = tune(),
                             trees = 1000,
                             mode = "classification") %>%
  set_engine("catboost")

model %>% tunable()

# set the workflow
tune_workflow <- workflow() %>%
  # add the recipe
  add_formula(Target ~ .) %>%
  # add the model
  add_model(model)
tune_workflow

# Get information on all possible tunable arguments in the
# defined workflow(model_wflow) and check whether or not they
# are actually tunable.
tune_args(tune_workflow)

CatBoost_params <-
  dials::parameters(
    min_n(), # min data in leaf
    tree_depth(range = c(4,10)), # depth
    # In most cases, the optimal depth ranges from 4 to 10.
    # Values in the range from 6 to 10 are recommended.
    learn_rate() # learning rate
  )

### Tune model parameters
set.seed(2020)
params <- dials::grid_max_entropy(
  CatBoost_params,
  size = 5
)
params

tic()
model_grid <- tune::tune_grid(
  tune_workflow,
  resamples = data_folds,
  grid      = params,
  metrics   = yardstick::metric_set(roc_auc, mcc),
  control   = control_grid(verbose = TRUE)
)
toc()
model_grid

# tuning results
model_grid %>%
  collect_metrics()

# pick best model
best_parameters <- model_grid %>%
  select_best("mcc")
best_parameters

# finalize wotkflow
final_model <-
  model %>%
  finalize_model(best_parameters) %>%
  fit(Target ~ ., data = df.train)
final_model

head(predict(final_model,
        new_data = df.test,
        type = "prob"))

# A tibble: 6 x 2
  .pred_No .pred_Yes
     <dbl>     <dbl>
1    4.86     -3.86 
2    4.31     -3.31 
3    1.20     -0.198
4    3.85     -2.85 
5    4.10     -3.10 
6    0.358     0.642 

R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.3.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8   
 [6] LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] doParallel_1.0.15     iterators_1.0.12      foreach_1.5.0         treesnip_0.1.0        tictoc_1.0            yardstick_0.0.7       workflows_0.2.0      
 [8] tune_0.1.1            tidyr_1.1.2           tibble_3.0.3          rsample_0.0.8         recipes_0.1.13        purrr_0.3.4           parsnip_0.1.3        
[15] modeldata_0.0.2       infer_0.5.3           ggplot2_3.3.2         dplyr_1.0.2           dials_0.0.9           scales_1.1.1          broom_0.7.0          
[22] tidymodels_0.1.1.9000

loaded via a namespace (and not attached):
 [1] nlme_3.1-148         matrixStats_0.56.0   lubridate_1.7.9      RColorBrewer_1.1-2   DiceDesign_1.8-1     repr_1.1.0           tools_4.0.2         
 [8] backports_1.1.10     utf8_1.1.4           R6_2.4.1             rpart_4.1-15         catboost_0.22        colorspace_1.4-1     nnet_7.3-14         
[15] withr_2.3.0          prettyunits_1.1.1    GGally_2.0.0         tidyselect_1.1.0     compiler_4.0.2       cli_2.0.2            checkmate_2.0.0     
[22] stringr_1.4.0        digest_0.6.25        ingredients_1.3.0    base64enc_0.1-3      pkgconfig_2.0.3      htmltools_0.5.0      DALEX_1.3.0         
[29] lhs_1.0.2            rlang_0.4.7          rstudioapi_0.11      pryr_0.1.4           generics_0.0.2       jsonlite_1.7.1       ModelMetrics_1.2.2.2
[36] magrittr_1.5         rapportools_1.0      Matrix_1.2-18        Rcpp_1.0.5           munsell_0.5.0        fansi_0.4.1          GPfit_1.0-8         
[43] lifecycle_0.2.0      furrr_0.1.0          stringi_1.5.3        pROC_1.16.2          MASS_7.3-51.6        plyr_1.8.6           grid_4.0.2          
[50] listenv_0.8.0        crayon_1.3.4         lattice_0.20-41      cowplot_1.0.0        splines_4.0.2        pander_0.6.3         summarytools_0.9.6  
[57] magick_2.4.0         knitr_1.30           pillar_1.4.6         tcltk_4.0.2          reshape2_1.4.4       codetools_0.2-16     stats4_4.0.2        
[64] glue_1.4.2           data.table_1.12.8    renv_0.12.0-12       vctrs_0.3.4          gtable_0.3.0         reshape_0.8.8        drifter_0.2.1       
[71] future_1.19.1        assertthat_0.2.1     xfun_0.17            gower_0.2.2          skimr_2.1.2          prodlim_2019.11.13   class_7.3-17        
[78] survival_3.1-12      timeDate_3043.102    lava_1.6.8           globals_0.13.0       ellipsis_0.3.1       caret_6.0-86         ipred_0.9-9

**Comment by Athospd** [comment]
Spot on! Thanks. There was missing type = "prob" for predict() fun. Fixed.
